src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -2097,21 +2097,29 @@ public void onWakeUp()
             homeBuilding.onWakeUp();
         }
 
-        final BlockPos spawn = Utils.scanForBlockNearPoint(
-          world,
-          getBedLocation(),
-          4,
-          2,
-          4,
-          PLAYER_HEIGHT,
-          Blocks.AIR,
-          Blocks.SNOW_LAYER,
-          Blocks.TALLGRASS,
-          Blocks.RED_FLOWER,
-          Blocks.YELLOW_FLOWER,
-          Blocks.CARPET);
-
-        if (spawn != null)
+        BlockPos spawn;
+        if (getBedLocation() != BlockPos.ORIGIN)
+        {
+            spawn = Utils.scanForBlockNearPoint(
+              world,
+              getBedLocation(),
+              4,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 4 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)